src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/HttpFuzzerMessageProcessorTagUIPanel.java

```diff
@@ -0,0 +1,148 @@
+    private static final String MATCH_BY_REGEX_NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.matchbyregex.name");
+    private static final String EXTRACT_BY_REGEX_LABEL_REGEX = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.extractbyregex.regex");
+    private static final String EXTRACT_BY_REGEX_NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.extractbyregex.name");
+    private final JPanel fieldsPanel;
+    private ButtonGroup radioGroup;
+
+    private ZapTextField matchByRegexFieldRegex;
+    private ZapTextField matchByRegexFieldTag;
+    private JRadioButton matchByRegexRadio;
+
+    private ZapTextField extractByRegexFieldRegex;
+    private JRadioButton extractByRegexRadio;
+
+    public HttpFuzzerMessageProcessorTagUIPanel() {
+        fieldsPanel = new JPanel();
+        radioGroup = new ButtonGroup();
+        GridLayout layout = new GridLayout(5, 2);
+        fieldsPanel.setLayout(layout);
+        createMatchByRegexControl();
+        createExtractByRegexControl();
+    }
+
+    private void createExtractByRegexControl() {
+        extractByRegexRadio = new JRadioButton();
+        addRadioButton(extractByRegexRadio, EXTRACT_BY_REGEX_NAME, false);
+
+        extractByRegexFieldRegex = new ZapTextField();
+        addLabelWithTextField(extractByRegexFieldRegex, EXTRACT_BY_REGEX_LABEL_REGEX);
+    }
+
+    private void createMatchByRegexControl(){
+        matchByRegexRadio = new JRadioButton();
+        addRadioButton(matchByRegexRadio, MATCH_BY_REGEX_NAME, true);
+
+        matchByRegexFieldRegex = new ZapTextField();
+        addLabelWithTextField(matchByRegexFieldRegex, MATCH_BY_REGEX_LABEL_REGEX);
+
+        matchByRegexFieldTag = new ZapTextField();
+        addLabelWithTextField(matchByRegexFieldTag, MATCH_BY_REGEX_LABEL_TAG);
+    }
+
+    private void addRadioButton(JRadioButton radio, String name, boolean selected) {
+        radio.setText(name);
+        radio.setSelected(selected);
+        fieldsPanel.add(radio);
+        fieldsPanel.add(new Panel());
+        radioGroup.add(radio);
+    }
+
+    private void addLabelWithTextField(JTextField textField, String labelName){
+        JLabel label = new JLabel(labelName,  JLabel.TRAILING);
+        fieldsPanel.add(label);
+        label.setLabelFor(textField);
+        fieldsPanel.add(textField);
+    }
+
+    @Override
+    public JPanel getComponent() {
+        return fieldsPanel;
+    }
+
+    @Override
+    public void setFuzzerMessageProcessorUI(HttpFuzzerMessageProcessorTagUI processorUI) {
+        TagRule rule = processorUI.getTagRule();
+        if(rule.getClass() == MatchByRegexTagRule.class){
+            MatchByRegexTagRule tagRule = (MatchByRegexTagRule)rule;
+            matchByRegexRadio.setSelected(true);
+            matchByRegexFieldRegex.setText(tagRule.getRegex());
+            matchByRegexFieldTag.setText(tagRule.getTag());
```

Comment:
The edits should be discarded ("discardAllEdits()") after setting the text to the ZapTextField(s), so that it does not allow to undo the text set.